deploy/lib/xquery/setup.xqy

```diff
@@ -4742,6 +4742,10 @@ declare function setup:validate-external-security(
 declare function setup:create-roles(
   $import-config as element(configuration))
 {
+  (: get the existing role names from the default security DB :)
+  let $existing-role-names := setup:get-existing-role-names()
+  return
```

Comment:
no need for a separate FLWOR; remove `return`. (FLWOR statement can mix let and for in any order)